src/main/flight/position.c

```diff
@@ -137,16 +142,40 @@ void calculateEstimatedAltitude(timeUs_t currentTimeUs)
     }
 #endif
 
-    if (ARMING_FLAG(ARMED) && !altitudeOffsetSet) {
+    if (ARMING_FLAG(ARMED) && !altitudeOffsetSetBaro) {
         baroAltOffset = baroAlt;
-        gpsAltOffset = gpsAlt;
-        altitudeOffsetSet = true;
-    } else if (!ARMING_FLAG(ARMED) && altitudeOffsetSet) {
-        altitudeOffsetSet = false;
+        altitudeOffsetSetBaro = true;
+    } else if (!ARMING_FLAG(ARMED) && altitudeOffsetSetBaro) {
+        altitudeOffsetSetBaro = false;
     }
+
     baroAlt -= baroAltOffset;
+
+    int goodGpsSats = 0;
+    int badGpsSats = -1;
+
+    if (haveBaroAlt) {
+        goodGpsSats = positionConfig()->altNumSatsGpsUse;
+        badGpsSats = positionConfig()->altNumSatsBaroFallback;
+    }
+
+    if (ARMING_FLAG(ARMED)) {
```

Comment:
Not sure if it's a good idea to catch some random altitude in the air. If user arms and fly without waiting GPS with no baro - let him see an absolute altitude instead. At least it will be consistent.